File: MenuProject/app/src/main/java/com/example/menuproject/MainActivity.java
package com.example.menuproject;

import android.content.Intent;
import android.net.Uri;
import android.os.Bundle;
import android.support.design.widget.FloatingActionButton;
import android.support.design.widget.Snackbar;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.Toolbar;
import android.view.View;
import android.view.Menu;
import android.view.MenuItem;
import android.widget.Button;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        Toolbar toolbar = findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);


        Button activity = findViewById(R.id.btnNewActivity);
        activity.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Intent i = new Intent(MainActivity.this, NewActivity.class);
                startActivity(i);
                finish();

            }
        });

        Button sms = findViewById(R.id.btnSms);
        sms.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                final Intent intent = new Intent(Intent.ACTION_SENDTO);
                intent.setData(Uri.parse("smsto:" + Uri.encode("2038758978")));
                intent.putExtra("sms_body", "John Pelliccia");
                startActivity(intent);

            }
        });

        Button call = findViewById(R.id.btnPhone);
        call.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                final Intent intent = new Intent(Intent.ACTION_DIAL);
                intent.setData(Uri.parse("tel:" + "2038758978"));
                startActivity(intent);

            }
        });

        Button web = findViewById(R.id.btnWeb);
        web.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse("http://www.macrumors.com"));
                startActivity(intent);

            }
        });

        Button maps = findViewById(R.id.btnMap);
        maps.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                String geoUri = String.format("geo:42.342824, -71.079408");
                Uri geo = Uri.parse(geoUri);
                final Intent intent = new Intent(Intent.ACTION_VIEW, geo);
                startActivity(intent);

            }
        });

        Button share = findViewById(R.id.btnShare);
        share.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {

                final Intent intent = new Intent(Intent.ACTION_SEND);
                intent.setType("text/plain");
                startActivity(Intent.createChooser(intent, "Share the love"));
            }
        });


    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_main, menu);
        return true;
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        // Handle item selection
        switch (item.getItemId()) {
            case R.id.action_help:
                Intent help = new Intent(this, HelpActivity.class);
                startActivity(help);
                return true;
            case R.id.action_settings:
                //showHelp();
                return true;
            default:

        }

        return super.onOptionsItemSelected(item);

    }


}

File: AndroidKotlin1.2/app/src/main/java/com/example/diceroller/MainActivity.kt
package com.example.diceroller

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import android.widget.Toast
import java.util.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val rollButton: Button = findViewById(R.id.roll_button)

        rollButton.setOnClickListener { rollDice() }

        val countButton: Button = findViewById(R.id.count_button)
        countButton.setOnClickListener { countIt() }

        val resetButton: Button = findViewById(R.id.reset_button)
        resetButton.setOnClickListener { reset() }

    }

    private fun rollDice() {
        val resultText: TextView = findViewById(R.id.result_text)
        val randomInt = Random().nextInt(6) + 1
        resultText.text = randomInt.toString()

       // resultText.text = "Dice Rolled!"
        //Toast.makeText(this, "button clicked", Toast.LENGTH_SHORT).show()

    }

    private fun countIt(){
        val resultText: TextView = findViewById(R.id.result_text)

        if(resultText.text == "Hello World!"){
            resultText.text = "1"
        }else{
            var result = resultText.text.toString().toInt()

            if(result <6){
                result++
                resultText.text = result.toString()
            }
        }
    }

    private fun reset(){
        val resultText: TextView = findViewById(R.id.result_text)
        resultText.text = "0"


    }
}
